frontend/app/utils/fetch-utils.ts

```diff
@@ -40,3 +41,24 @@ export function getFetchFn(proxyUrl?: string, timeout?: number) {
   log.debug('No proxy configured; using global fetch');
   return global.fetch;
 }
+
+/**
+ * Wraps a fetch() function and adds instrumentation/metrics collection for error handling.
+ *
+ * @param fetchFn An existing fetch() function used to make the actual request
+ * @param metricPrefix A string prefix used for naming metrics (ex. "http.client.example-api.example-endpoint.gets", "http.client.example-api.example-endpoint.posts").
+ * @param input The URL or request object specifying the target of the fetch request
+ * @param init An object containing additional options for the fetch request (ex. headers, method)
+ * @returns A Promise that resolves to the Response object from the fetch() call
+ * @throws The original error thrown by the underlying fetch() call
+ */
+export async function instrumentedFetch(fetchFn: FetchFunction, metricPrefix: string, input: string | URL, init?: FetchFunctionInit) {
+  try {
+    const response = await fetchFn(input, init);
+    return response;
```

Comment:
I think you should consider looking at the response status here and emitting a metric if it's OK or not.

Reply:
Considered and added [here](https://github.com/DTS-STN/canadian-dental-care-plan/compare/feb61527a9b1e8d2be2c21c326302b33115cb5db..dfa6ac08b8ca8eb92cb7d1f713dc6e4d3d7b966f#diff-5f5669eb63caa505348a0900e4f6e7d3ec8f675ea39ebf957d511dccaedcdf4aR59).